README.md

```diff
@@ -165,3 +167,68 @@ definition location: /Volumes/src/orc/extras/struct0/src/a.cpp:3
 ```
 
 What line and file the object was declared in. So line 3 of `a.cpp` in this example.
+
+# The ORC Test App (`orc_test`)
+
+To ensure that ORC is catching what is purports to catch, a unit test application has been added to the system. `orc_test` introduces a miniature "build system" to generate object files from known sources to produce known ODR violations. It then processes the object files using the same engine as the ORC command line tool, and compares the results against an expected ODRV report list.
+
+## The Test Battery Structure
+
+Every unit test in the battery is discrete, and contains:
+
+1. A set of source files
+2. `odrv_test.toml`, a high level TOML file describing the parameters of the test
+
+In general, a single test should elicit a single ODR violation, but this may not be possible in all cases. 
+
+### The Source File(s)
+
+These files are standard C++ source files. Their quantity and size should be very small - only big enough as needed to cause the intended ODRV.
+
+### The `odrv_test.toml` File
+
+The settings file describes to the test application what source(s) need to be compiled, what compilation flags should be used for the test, and what ODRVs the system needs to be on the lookout as a result of linking the generated object file(s) together.
+
+#### Specifying Sources
+
+Test sources are specified with a `[[source]]` directive:
+
+```toml
+[[source]]
+    path = "one.cpp"
+    obj = "one"
+    flags = [
+        "-Dfoo=1"
+    ]
+```
+
+The `path` field describes a path to the file relative to `odrv_test.toml`. It is the only required field.
+
+The `obj` field specifies the name of the (temporary) object file to be created. If this name is omitted, a pseudo-random name will be used.
+
+The `flags` field specifies compilation flags that will be used specifically for this compilation unit. Using this field, it is possible to reuse the same source file with different compilation flags to elicit an ODRV.
+
+#### Specifying ODRVs
+
+ODRVs are specified with the `[[odrv]]` directive:
+
+
+```toml
+[[odrv]]
+    category = "subprogram:vtable_elem_location"
+    linkage_name = "_ZNK6object3apiEv"
+```
+
+The `category ` field describes the specific type of ODR violation the test app should expect to find.
+
+The `linkage_name` field describes the specific symbol that caused the ODRV. It is currently unused, but will be enforced as the test app matures.
+
+#### Other fields
```

Comment:
Should "Other fields" be "Fields In Progress" ? "In Development"? Not sure. Just a thought.

Reply:
I went with "Fields In Development"